frontend/catalyst/jax_tracer.py

```diff
@@ -1238,3 +1242,12 @@ def check_full_raise(arr, func):
         # TODO: `check_jaxpr` complains about the `AbstractQreg` type. Consider fixing.
         # check_jaxpr(jaxpr)
     return closed_jaxpr, out_type, out_tree, return_values_tree
+
+
+def _add_mlir_quantum_decomposition(f, device):
+    # TODO: make this non related to the name of the device
+    if device.original_device.name == "oqd.cloud":
+        apply_registered_pass_p.bind(
+            pass_name="ions-decomposition",
+            options=f"func-name={f.__name__}",
```

Comment:
This is more organizational, but I'm wondering if we should stick this pass into the `catalyst/passes.py` module, since that's where the other passes live as well?

Reply:
I have moved the function to the passes file